pkg/vault/vault.go

```diff
@@ -87,58 +89,99 @@ func (v *vaultApi) getClient() (*api.Client, error) {
 	return api.NewClient(config)
 }
 
-func (v *vaultApi) tokenTTL() (int64, error) {
-	resp, err := v.Request(http.MethodGet, SelfTokenURL, nil)
-	if err != nil || resp == nil {
-		return 0, fmt.Errorf("couldn't obtain self token information: %v", err)
+// A token is considered invalid if we receive 400 status codes
+func (v *vaultApi) tokenTTL() (ttl int64, invalid bool, err error) {
+	s, resp, err := v.Request(http.MethodGet, SelfTokenURL, nil)
+	if resp != nil && (resp.StatusCode == 400 || resp.StatusCode == 403) {
+		return 0, true, err
 	}
-	ttlNumber, ok := resp.Data["ttl"].(json.Number)
+	if err != nil {
+		return 0, false, fmt.Errorf("couldn't obtain self token information: %v", err)
+	}
+	ttlNumber, ok := s.Data["ttl"].(json.Number)
 	if !ok {
-		return 0, fmt.Errorf("couldn't obtain token TTL")
+		return 0, false, fmt.Errorf("couldn't obtain token TTL")
+	}
+	ttl, err = ttlNumber.Int64()
+	return ttl, false, err
+}
+
+// A token is not considered renewable if we receive a 400 error
+// for any other case we consider that it can still be renewed
+// and we are having problems connecting to the server.
+func (v *vaultApi) renewToken() (renewable bool, err error) {
+	s, resp, err := v.Request(http.MethodPost, SelfTokenRenewURL, nil)
+	if resp != nil && (resp.StatusCode == 400 || resp.StatusCode == 403) {
+		return false, err
+	}
+	if s != nil && s.Auth != nil {
+		return s.Auth.Renewable, err
+	}
+	if s != nil {
+		return s.Renewable, err
 	}
-	return ttlNumber.Int64()
+	return true, err
 }
 
 func (v *vaultApi) autoRenewToken() {
-	var ttl int64
-	var err error
+	const (
```

Comment:
If they are constants, should'n be at the top of the file

Reply:
They are constants, but they only make sense in the scope of this function, I don't think it worths to define them package-wide.

Reply:
And what about just initializing a map with the values needed?